### ISSUE TITLE
AppData Editor and Wolf Link

### ISSUE THREAD

**Issue by Athius** [AcK77/AmiiBomb-uino#6]
Hi,

I just try the new AppData Editor on Wolf Link Amiibo dump but when I create the NTAG and try it on Zelda BTW, nothing appends (no more hearts). Is it a normal?

Thanks for your great job!

**Comment by jokkovitch** [comment]
What do you mean by "No more hearts" ? 
Did you summon the wolf ?

**Comment by Athius** [comment]
I summon the wolf but he still has 3 hearts instead of 20 (the value I set in AppData Editor). However, when I dump the NTAG previously created and I see 20 hearts in AppData Editor.

**Comment by AcK77** [comment]
Did you set up the level of the Wolf at the higher? It's strange because I try the editor on many Wolf Dump and summon them in Zelda BTW and all works... The number of Wolf Hearth depend to the Level you achieve in Twilight Princess so you have to maxed out it too!

**Comment by Athius** [comment]
Yeah, I set up the level of the Wolf to level 40. Maybe I'm not far enough in BTW?

**Comment by jokkovitch** [comment]
Did you flash it to a new ntag ? or did you try to flash it again to the same one ?

**Comment by Athius** [comment]
On a new tag. I tried to the same one but of course, an error occured (the ntag is locked).

**Comment by AcK77** [comment]
Can you provide me your NTAG dump by mail? Before and after the edit?
Maybe i forgot something, but to know if it's true, I need different dump... All dump I have works after edit them...

**Comment by peacepenguin** [comment]
I don't think there's a bug in the appdata editor, I believe there is just a situation where the appdata editor "can't" work based on the state of the dump being modified:

I think the problem here is that you used a "blank" wolf link BIN to flash initially. Using the AppData editor is technically working fine, but there isn't already a "complete" AppData from an actual Wii U Twilight Princess game save on your tag. So the AppData modification isn't even read since BOTW game doesn't detect the rest of the Twilight Princess save data on the Amiibo, so it gives up from the start, and only provides 3 hearts. Just like if you got a brand new wolf link amiibo and tried to use it with BOTW, only 3 hearts.

The trick to overcome it without wasting another blank ntag is this, and the functionality is built into AmiiBomb already:
"Aquire" a Wolf Link.bin that is known to be working with 20 hearts from the web, extract its entire AppData using Amiibomb, and write that entire Appdata to your wolf link card/tag that only gives 3 hearts. Then wait 24 hours! The game will provide the same wolf stats as the first one scanned in a 24 hour period. Even with a "real" amiibo if i scan a 3 heart wolf, then later the same day a legit working 20 heart wolf amiibo, you'll still only get a 3 heart wolf spawned.

This will work because the appdata is not write protected, so you can use amiibomb in the same way as the "powersaves" accessory to read/write the Appdata section of even real amiibos. The problem you ran into was trying to re-write the entire Dump file, not just the Appdata section. There is a different option within amiibomb for writing only the appdata section.

The editor comes into play in this situation:
Use a real wolf link amiibo and play twilight princess cave mini game and beat it, save the results to the real wolf link amiibo. Obviously this save won't have 20 hearts if you haven't 100% twilight princess.
Dump this real amiibo to amiibomb
Use appdata editor to modify the hearts to 20
Write the modified dump to a new ntag
The new wolf link ntag will work fine with 20 hearts since it has a "normal" and THEN modified appdata save.

-BenD

**Comment by peacepenguin** [comment]
I don't think there's a bug in the appdata editor, I believe there is just a situation where the appdata editor "can't" work based on the state of the dump being modified:

I think the problem here is that you used a "blank" wolf link BIN to flash initially. Using the AppData editor is technically working fine, but there isn't already a "complete" AppData from an actual Wii U Twilight Princess game save on your tag. So the AppData modification isn't even read since BOTW game doesn't detect the rest of the Twilight Princess save data on the Amiibo, so it gives up from the start, and only provides 3 hearts. Just like if you got a brand new wolf link amiibo and tried to use it with BOTW, only 3 hearts.

The editor comes into play in this situation:

new amiibo figure wolf link with "blank" appdata -> played cave mini game (owner data, and heart qty now on Amiibo) -> dump to bin -> modified bin appdata amiibomb -> write this to new NTAG.

I think there should be a way to add functionality to amiibomb to be able to write the appdata portion only to a write-protected ntag with an amiibo on it or even an actual amiibo. Similar to how the "powersaves" accessory works. This would allow you to "correct" your 3 heart wolf without using another tag. Or even to upload someone else's 20 heart save data.

Perhaps someone else's app data could be added to the editor itself so it detects if any personalized data exists on the amiibo, and if not, put one there that it can then be successfully modified. Otherwise a warning could be shown that indicates the appdata editor won't work when trying to edit a "blank" wolf link amiibo.

**Comment by Athius** [comment]
@AcK77 Did you recieve my message? I sent it few weeks ago with the dumps

**Comment by peacepenguin** [comment]
@Athius  Can you upload the files here into this comment system?
Maybe setup a repo on github to upload the dumps into?
